.github/workflows/linux.yml

```diff
@@ -24,56 +24,81 @@ on:
+  group: ${{ github.ref || github.run_id }}-linux
   cancel-in-progress: true
 
 jobs:
   Build:
     defaults:
       run:
         shell: bash
-    runs-on: ubuntu-20.04-8-cores
+    runs-on: aks-linux-16-cores
+    container:
+      image: openvinogithubactions.azurecr.io/dockerhub/ubuntu:20.04
+      volumes:
+        - /mount/caches:/mount/caches
     env:
+      DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
       CMAKE_BUILD_TYPE: 'Release'
       CMAKE_GENERATOR: 'Ninja'
       CMAKE_CXX_COMPILER_LAUNCHER: ccache
       CMAKE_C_COMPILER_LAUNCHER: ccache
-      OPENVINO_REPO: ${{ github.workspace }}/openvino
-      OPENVINO_CONTRIB_REPO: ${{ github.workspace }}/openvino_contrib
-      INSTALL_DIR: ${{ github.workspace }}/install
-      INSTALL_TEST_DIR: ${{ github.workspace }}/install/tests
-      SAMPLES_INSTALL_DIR: ${{ github.workspace }}/install/samples
-      LAYER_TESTS_INSTALL_DIR: ${{ github.workspace }}/install/tests/layer_tests
-      MODEL_HUB_TESTS_INSTALL_DIR: ${{ github.workspace }}/install/tests/model_hub_tests
-      BUILD_DIR: ${{ github.workspace }}/build
-      OV_TEMP: ${{ github.workspace }}/openvino_temp
+      GITHUB_WORKSPACE: '/__w/openvino/openvino'
+      OPENVINO_REPO: /__w/openvino/openvino/openvino
+      OPENVINO_CONTRIB_REPO: /__w/openvino/openvino/openvino_contrib
+      INSTALL_DIR: /__w/openvino/openvino/install
+      INSTALL_TEST_DIR: /__w/openvino/openvino/install/tests
+      SAMPLES_INSTALL_DIR: /__w/openvino/openvino/install/samples
+      LAYER_TESTS_INSTALL_DIR: /__w/openvino/openvino/install/tests/layer_tests
+      MODEL_HUB_TESTS_INSTALL_DIR: /__w/openvino/openvino/install/tests/model_hub_tests
+      BUILD_DIR: /__w/openvino/openvino/build
+      DATA_PATH: /__w/openvino/openvino/testdata
+      MODELS_PATH: /__w/openvino/openvino/testdata
+      OV_TEMP: /__w/openvino/openvino/openvino_temp
       PYTHON_STATIC_ARGS: -m "not dynamic_library"
+      CCACHE_DIR: /mount/caches/ccache/ubuntu22_x86_64_Release
+      CCACHE_TEMPDIR: /__w/openvino/openvino/ccache_temp
+      CCACHE_MAXSIZE: 50G
     steps:
+      - name: Install git
+        run: |
+          apt-get update
+          apt-get install -y git git-lfs
+
       - name: Clone OpenVINO
         uses: actions/checkout@v4
         with:
-          path: 'openvino'
-          submodules: 'true'
+          path: ${{ env.OPENVINO_REPO }}
+          submodules: 'recursive'
 
       - name: Clone OpenVINO Contrib
         uses: actions/checkout@v4
         with:
           repository: 'openvinotoolkit/openvino_contrib'
-          path: 'openvino_contrib'
+          path: ${{ env.OPENVINO_CONTRIB_REPO }}
+          submodules: 'recursive'
+
+      - name: Clone testdata for C API tests
+        uses: actions/checkout@v4
+        with:
+          repository: 'openvinotoolkit/testdata'
+          path: 'testdata'
+          submodules: 'recursive'
+          lfs: 'true'
```

Comment:
please, remove `openvinotoolkit/testdata` clone. 
It's removed in current master